src/OpenTelemetry/Metrics/ExplicitBucketHistogramConfiguration.cs

```diff
@@ -30,7 +30,7 @@ public class ExplicitBucketHistogramConfiguration : MetricStreamConfiguration
         /// Requirements:
         /// <list type="bullet">
         /// <item>The array must be in ascending order with distinct
-        /// values.</item>
+        /// values. double.NaN is not allowed</item>
```

Comment:
What about other IEEE754 subnormal (e.g. -Inf)?

Reply:
Good callout. I could do a follow up change for other subnormals.
@reyang @alanwest How should we treat subnormals being used as bounds or values.
I assume bounds, we should throw exception. But for values, there's decisions to be made (SDK specific) of whether we should have them affect bucket counts (and which bucket), sum, min, and max for histograms. But I guess this also expands to the other instrument types.

Reply:
Check these:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#numerical-limits-handling

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exponential-histogram-aggregation-handle-all-normal-values

Reply:
The 2nd link applies to exponential histogram, but many ideas might be helpful for explicit bucket histogram, too.

Reply:
Nice. Hadn't seen the new wording for exponential histograms. I think it makes sense to follow this for explicit-bound histograms too. That is, effectively drop all sub-normal values from affecting min, max, and sum. I also think this wording would suggest that it is ok to not modify bucket counts either:

> because these values do not map into a valid bucket

Reply:
Sharing some additional context/thoughts:

![image](https://user-images.githubusercontent.com/17327289/168403535-531dfe52-d1ff-4f36-ad60-1f6a20962c9c.png)

Reply:
I think for Sum, the challenge is that even if we don't take subnormal as input, the output could still be subnormal:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/d983cb1270fa7a2738f9179d46f581268a4bb449/test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs#L176-L186